dotCMS/src/main/java/com/dotcms/experiments/business/ExperimentUrlPatternCalculator.java

```diff
@@ -0,0 +1,86 @@
+package com.dotcms.experiments.business;
+
+import com.dotmarketing.business.APILocator;
+import com.dotmarketing.exception.DotDataException;
+import com.dotmarketing.portlets.htmlpageasset.model.HTMLPageAsset;
+import com.dotmarketing.portlets.structure.StructureUtil;
+import com.dotmarketing.util.UtilMethods;
+import java.util.ArrayList;
+import java.util.List;
+
+/**
+ * Util class to calculate the regex pattern for a given {@link HTMLPageAsset}
+ */
+public enum ExperimentUrlPatternCalculator {
```

Comment:
This thing does not looks like enum, I guess it is just to ensure the singleton but we have another ways to do it

Reply:
Yeah Enums are the recommended way in books like Effective Java by Joshua Blotch for implementing singletons.

Reply:
ok